README.md

```diff
@@ -341,10 +341,14 @@ d.memory["main_arena"] = b"12345678"
 
 `wait()` will block the execution of the Python script until the debugging process interrupts.
 
-`finish([exact=True])` will continue the execution until the current function returns or the process is stopped (e.g., breakpoint).
+`finish([heuristic="backtrace"])` will continue the execution until the current function returns or the process is stopped (e.g., breakpoint).
 
-In **exact** mode, the debugger will iteratively perform single hardware steps until the return instruction of the current frame is executed. In **non-exact** mode, a breakpoint is placed on the return address and execution continues. In most cases, when the function frame is aligned with the calling convention, the flow of both modes coincides. However, in case of packed or non-conventional binaries, it is possible that the address found on the stack is wrong or cannot be detected by a software breakpoint. The reason why non-exact mode could be preferred is speed, as allowing execution to continue is still faster than hardware stepping for big functions.
+The available heuristics are:
 
+- **backtrace**: This heuristic uses the saved return address found on the stack or on a dedicated register to find the return address of the current function. A breakpoint is applied to the resolved address and execution is continued. This is the fastest heuristic and is fairly reliable, but it may not work in the presence of self-modifying code.
+- **step-mode**: This heuristic steps one instruction at a time until the ret instruction is executed in the current frame (nested calls are handled). This is a reliable heuristic, but is slow and fails in the case of tailcalls or similar optimizations.
+
```

Comment:
It is not very clear to me why the `step-mode` should break in the case of tailcalls and the `backtrace`  mode should not. In a tailcall, we have a jump instead of a call to execute the tail function. The callee will use the caller's stack frame, including the return address. Sooner or later, a return referencing that return address will be executed. We might lose  precision with, for example, recursive tailcalls, it is true, but this should affect both `backtrace` and `step-mode`. Why is this not the case?

Reply:
You are correct, I meant the step-mode should break in the case of **internal**, nested tailcalls, since the call counter will not be increased. The backtrace is not affected by this behaviour. Instead, if the tailcall is the "origin" of the function, both will break.

Reply:
Ok, but I mean, even if you do not increment the counter, it should not be a problem.

Let's consider I call finish to function A. Function A then contains a tailcall with a jump, and this tailcall is recursive. At the end of the recursion, the very last function (tail)called should return. Since the stack frame is the same as A due to tailcall optimization, it will return to A's caller, as if no extra calls were performed. Hence, the counter should not be a problem in this case (it is neither incremented nor decremented, and should be balanced), and the behavior should be coherent with both heuristics.

Obviously, problems will arise in case of finish inside the tailcall (if you want to stop at the end of a specific tailcall instance). But it is a problem for both heuristics.